contracts/WETH9.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+/**
+ * Copyright (C) 2015, 2016, 2017 Dapphub
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * Imported as at commit 33d01d471437e1ab6861e4545ea4bb3895fd4d74 from:
+ * UMAprotocol/protocol/packages/core/contracts/financial-templates/common/WETH9.sol
+ * Updated with the following changes post-import:
+ * - Marked transferFrom() as virtual
+ */
+
+pragma solidity ^0.8.0;
+
+contract WETH9 {
```

Comment:
nit: we probably don't want two contracts in the repo called WETH9 -- this causes issues in hardhat I think. Any reason we can't just delete the interface or rename it to WETH9Interface?

Reply:
Good point - I'd totally missed the existing interface. I've split this out - see https://github.com/across-protocol/contracts-v2/pull/227.